std/math.d

```diff
@@ -249,8 +249,504 @@ enum RealFormat
     ieeeQuadruple,
 }
 
-// Constants used for extracting the components of the representation.
-// They supplement the built-in floating point properties.
+/* Low-level access to the individual parts of a floating-point value.
+
+   TODO: Make this publically available through std.bitmanip once the API
+   is stable and tested.
+*/
+union RealRep(N)
+if (isFloatingPoint!N &&  is(N == Unqual!N))
+{
+pure: @safe: nothrow: @nogc:
+pragma(inline, true):
+
+    // The floating-point number who's bit fields will be exposed.
+    N num = N.init;
+    alias num this;
+    this(const N num) inout
+    {
+        this.num = num;
+    }
+
+    static if (N.mant_dig == 106 && N.max_exp == 0x400)
+    {
+        /* N has IBM's double + double format, which is actually two ieeeDouble
+        (see below) stuck together. Pretending otherwise is not helpful at this
+        level. */
+        enum format = RealFormat.ibmExtended;
+
+        struct
+        {
+            version(LittleEndian)
+            {
+                RealRep!double lo;
+                RealRep!double hi;
+            }
```

Comment:
After some testing, can safely say that this is wrong. The most significant part is always first, regardless of of endian-ness on ibmExtended.